derive/Cargo.toml

```diff
@@ -1,12 +1,12 @@
 [package]
 name = "parity-scale-codec-derive"
 description = "Serialization and deserialization derive macro for Parity SCALE Codec"
-version = "3.6.8"
-authors = ["Parity Technologies <admin@parity.io>"]
-license = "Apache-2.0"
-edition = "2021"
-rust-version = "1.56.1"
```

Comment:
We do also bump the min rust version here, not sure if that is considered a breaking change? Maybe stay with the old version - if possible.

Reply:
Out of interest, I used `cargo msrv` to find the actual minimum supported Rust version, and it's currently actually 1.79 owing to a dependency using more modern features.

If we care about honouring a specific MSRV we should probably add `cargo msrv` to CI as eg a daily check that we are compliant, but my view is not to worry too much because any dependency that increases its MSRV in a non breaking release will force it on us all of a sudden anyway, unless we pin everything!

Anyway, in this PR I'll set it to 1.79 to align with reality!

Reply:
Yea thats fine. I thought that we already checked it in CI, but apparently not 😆

Reply:
Just reporting that this breaks dependency requirement for cargo.

https://doc.rust-lang.org/cargo/reference/specifying-dependencies.html#specifying-dependencies-from-cratesio

```
3.6  :=  >=3.6.0, <4.0.0
```

Reply:
> Just reporting that this breaks dependency requirement for cargo.

Sorry; what do you mean by this? What specifically breaks?

Reply:
in our crates, we were using rust-version 1.75 with parity-scale-codec 3.6.12. after 3.7.0 release, our msrv builds are failing.

you should have released 4.0.0.

Reply:
Ah I see! 

It's up to the crate to determine whether MSRV breaks are indeed a breaking semver change or not. The general consensus in the Rust community is slightly torn but leans towards MSRV bumps _not_ requiring a major SemVer bump in crates.

In the case of this crate, it turned out that the actual Rust version supported prior to this release was already much higher than the reported MSRV (owing to a dependency havign a higher MSRV than we did). So by bumping the MSRV in this crate we're really just reflecting reality better!

In general, MSRV changes happen a bunch in minor and patch releases in transitive dependencies, making it very difficult to guarantee that this crate conforms to any particular MSRV short of pinning our dependency versions. This makes it very difficult to guarantee anything re MSRV in general.